includes/Modules/Tag_Manager.php

```diff
@@ -466,8 +466,27 @@ protected function create_data_request( Data_Request $data ) {
 					/* translators: %s: Missing parameter name */
 					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'containerID' ), array( 'status' => 400 ) );
 				}
-				return function() use ( $data ) {
-					$this->get_settings()->merge( array( 'containerID' => $data['containerID'] ) );
+
+				$usage_context        = $data['usageContext'] ?: self::USAGE_CONTEXT_WEB;
+				$valid_usage_contexts = array_keys( $this->context_map );
+
+				if ( ! in_array( $usage_context, $valid_usage_contexts, true ) ) {
```

Comment:
I think this could use `isset( $this->context_map[ $usage_context ] )` for slightly better performance over `in_array` - especially since this is the original state of the array anyway. It's super minor, but I think the `array_keys` transformation only needs to be done in the clause for imploding them.

Reply:
Sure, I've made that change. I think it was maybe a tiny bit more explicit/expressive before but not a big deal. I think `$data` validation in general could be improved a lot. Most of them are simple checks which are be handled out-of-the-box by core's handling using the route definition if these were normal WP REST routes. Would be a huge improvement if we could find a way to leverage core rest routes in a way that supports batch requests. Do you think this makes sense to look into as part of #166 ?

Reply:
Let's leave #166 as it is for now, to just move the routes. Regarding batch request support, I don't think we should invest in that because it's likely we will drop support for them in the future. If not, we can revisit then.